keps/159-scheduler-simulator-operator/README.md

```diff
@@ -0,0 +1,118 @@
+# KEP-159: scheduler simulator operator
+
+## Summary
+
+A new custom resource to represent a simulator itself and a custom controller to manage it are introduced.
+
+## Motivation
+
+With this, we can easily create/use the simulator as sandbox for the simulator by creating the Simulator custom resource. 
+
+This feature also makes it easier to handle simulators from other CRD or controllers. 
+They can gets the information for the simulator easily by accessint the `Simulator` resource.
+
+### Goals
+
+- Users can define simulators with Simulator CRD.
+- The controller will manage(create, edit, delete) simulators.
+
+### Non-Goals
+
+Create new `Simulator` from Web UI. (may be implemented in the future, but out of scope of this proposal.)
+
+## Proposal
+
+### Implementation design details
+
+#### Required prior knowledge
+
+The current simulator is built in a single binary: the binary contains kube-apiserver, kube-scheduler and the simulator backend (which provides the web UI specific feature.)
+
+And, the simulator can be configured via environment variables
+
+#### Simulator CRD
+
+The CRD `Simulator` will be applied to kube-apiserver in user's cluster. (not to simulator's kube-apiserver.) 
+
+It will create the Pod for the simulator in the same namespace.
+
+```go
+type Simulator struct {
+  metav1.TypeMeta 
+  metav1.ObjectMeta // namespaced.
+
+  Spec SimulatorSpec
+  Status SimulatorStatus
+}
+
+// SimulatorSpec has the spec for the simulator.
+// All fields can be updated (but a Pod may be re-created.)
+type SimulatorSpec struct {
+  // KubeAPIServerPort indicates kube-apiserver’s URL which the simulator has internally.
+  // (The each simulator has own kube-apiserver internally.)
+  // This field's value will be added to a "simulator" container's env as "KUBE_API_PORT".
+  KubeAPIServerPort            int 
```

Comment:
What is the `KubeAPIServerPort ` indicate? Port(int) or URL(string)?

Reply:
It's Port.

Reply:
Ooh, okay